judge/views/two_factor.py

```diff
@@ -0,0 +1,236 @@
+        return JsonResponse(data, encoder=WebAuthnJSONEncoder)
+
+    def post(self, request, *args, **kwargs):
+        if not request.session.get('webauthn_attest'):
+            return HttpResponseBadRequest()
+
+        if 'credential' not in request.POST or len(request.POST['credential']) > 65536:
+            return HttpResponseBadRequest(_('Invalid WebAuthn response'))
+
+        if 'name' not in request.POST or len(request.POST['name']) > 100:
+            return HttpResponseBadRequest(_('Invalid name'))
+
+        credential = json.loads(request.POST['credential'])
+
+        response = webauthn.WebAuthnRegistrationResponse(
+            rp_id=settings.WEBAUTHN_RP_ID,
+            origin='https://' + request.get_host(),
+            registration_response=credential['response'],
+            challenge=request.session['webauthn_attest'],
+            none_attestation_permitted=True,
+        )
+
+        try:
+            credential = response.verify()
+        except Exception as e:
+            return HttpResponseBadRequest(str(e))
+
+        model = WebAuthnCredential(
+            user=request.profile, name=request.POST['name'],
+            cred_id=credential.credential_id.decode('ascii'),
+            public_key=credential.public_key.decode('ascii'),
+            counter=credential.sign_count,
+        )
+        model.save()
+
+        if not request.profile.is_webauthn_enabled:
+            request.profile.is_webauthn_enabled = True
+            request.profile.save(update_fields=['is_webauthn_enabled'])
+        return HttpResponse('OK')
+
+
+class WebAuthnAttestView(WebAuthnView):
+    def get(self, request, *args, **kwargs):
+        challenge = os.urandom(32)
+        request.session['webauthn_assert'] = webauthn_encode(challenge)
+        data = webauthn.WebAuthnAssertionOptions(
+            [credential.webauthn_user for credential in
+             request.profile.webauthn_credentials.select_related('user__user')],
+            challenge,
+        ).assertion_dict
+        return JsonResponse(data, encoder=WebAuthnJSONEncoder)
+
+
+class WebAuthnDeleteView(SingleObjectMixin, WebAuthnView):
+    def get_queryset(self):
+        return self.request.profile.webauthn_credentials.all()
+
+    def post(self, request, *args, **kwargs):
+        credential = self.get_object()
+        count = self.get_queryset().count()
+
+        if settings.DMOJ_REQUIRE_STAFF_2FA and self.request.user.is_staff and \
+                count <= 1 and not request.profile.is_totp_enabled:
+            return HttpResponseBadRequest(_('Staff may not disable 2FA'))
+        credential.delete()
+
+        if count <= 1:
+            request.profile.is_webauthn_enabled = False
+            request.profile.save(update_fields=['is_webauthn_enabled'])
+
+        return HttpResponse()
+
+
+class TwoFactorLoginView(SuccessURLAllowedHostsMixin, TOTPView):
+    form_class = TwoFactorLoginForm
+    title = _('Perform Two Factor Authentication')
```

Comment:
```suggestion
    title = _('Perform Two-Factor Authentication')
```

Reply:
`Two-Factor` has incorrect capitalization. Also this is out of scope because it was actually from `judge/views/totp.py`. Somehow it was detected as a new file instead of rename.